File: leetcode0070/solution.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let (mut prev, mut current) = (1, 1);
        for _ in 2..=n {
            (prev, current) = (current, prev + current);
        }

        current
    }
}

// impl Solution {
//     pub fn climb_stairs(n: i32) -> i32 {
//         let mut prev = 1;
//         let mut current = 1;

//         for i in 2..=n {
//             let tmp = current;
//             current = prev + current;
//             prev = tmp;
//         }

//         current
//     }
// }

File: leetcode0070/solution.cpp
/*
Solution 1:

DP

Time: O(n) | Space: O(1)
*/

class Solution {
public:
    int climbStairs(int n) {
        int a = 1;
        int b = 1;

        for (int i = 2; i <= n; ++i) {
            int tmp = a + b;
            a = b;
            b = tmp;
        }

        return b;
    }
};